orca/2.1-a-first-look-at-a-neural-network.ipynb

```diff
@@ -0,0 +1,518 @@
+    "\n",
+    "OrcaContext.log_output = False  # (True: display terminal's stdout and stderr in the Jupyter notebook).\n",
+    "cluster_mode = \"local\"\n",
+    "\n",
+    "if cluster_mode == \"local\":\n",
+    "    init_orca_context(cluster_mode=\"local\", cores=4)  # run in local mode\n",
+    "elif cluster_mode == \"k8s\":\n",
+    "    init_orca_context(cluster_mode=\"k8s\", num_nodes=2, cores=2)  # run on K8s cluster\n",
+    "elif cluster_mode == \"yarn\":\n",
+    "    init_orca_context(cluster_mode=\"yarn-client\", num_nodes=2, cores=2)  # run on Hadoop YARN cluster"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After importing the network configurations from Keras, we can call the `fit` method to fit the model to its training data. The model can be saved using the `save` method."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "tags": [
+     "outputPrepend"
+    ]
+   },
+   "outputs": [
+    {
+     "output_type": "stream",
+     "name": "stderr",
+     "text": [
+      "177/187 [===========================>..] - ETA: 0s - loss: 0.0250 - accuracy: 0.9919\n",
+      "178/187 [===========================>..] - ETA: 0s - loss: 0.0249 - accuracy: 0.9919\n",
+      "179/187 [===========================>..] - ETA: 0s - loss: 0.0250 - accuracy: 0.9919\n",
+      "180/187 [===========================>..] - ETA: 0s - loss: 0.0249 - accuracy: 0.9919\n",
+      "181/187 [============================>.] - ETA: 0s - loss: 0.0248 - accuracy: 0.9920\n",
+      "182/187 [============================>.] - ETA: 0s - loss: 0.0249 - accuracy: 0.9919\n",
+      "183/187 [============================>.] - ETA: 0s - loss: 0.0248 - accuracy: 0.9920\n",
+      "184/187 [============================>.] - ETA: 0s - loss: 0.0248 - accuracy: 0.9920\n",
+      "185/187 [============================>.] - ETA: 0s - loss: 0.0247 - accuracy: 0.9920\n",
+      "186/187 [============================>.] - ETA: 0s - loss: 0.0246 - accuracy: 0.9921\n",
+      "187/187 [==============================] - ETA: 0s - loss: 0.0245 - accuracy: 0.9921\n"
+     ]
+    },
+    {
+     "output_type": "execute_result",
+     "data": {
+      "text/plain": [
+       "'/tmp/mnist_keras.ckpt'"
+      ]
+     },
+     "metadata": {},
+     "execution_count": 13
+    },
+    {
+     "output_type": "stream",
+     "name": "stdout",
+     "text": [
+      "187/187 [==============================] - 15s 82ms/step - loss: 0.0245 - accuracy: 0.9921 - val_loss: 0.0353 - val_accuracy: 0.9883\n"
+     ]
+    }
+   ],
+   "source": [
+    "from zoo.orca.learn.tf2 import Estimator\n",
+    "\n",
+    "est = Estimator.from_keras(model_creator=model_creator, config={\"batch_size\": batch_size}, workers_per_node=1)\n",
+    "est.fit(data=train_data_creator,\n",
+    "        epochs=max_epoch,\n",
+    "        batch_size=batch_size,\n",
+    "        steps_per_epoch=60000 // batch_size,\n",
```

Comment:
@yangw1234 Can users not specify steps_per_epoch and validation_steps by default in case they want to train and validate the whole dataset? Manual calculation seems not user friendly.

Reply:
Yeah, I agree. However, in some versions of tensorflow if we did not specify the steps_per_epoch and the total number of samples is not a multiple of batch size, it will hang.

Reply:
OK. So for the time being users have to input this value right?